web/js/doc.js

```diff
@@ -99,12 +99,12 @@ window.onscroll = (event) => {
 
           if (linkable && clickable) {
             pre.classList.add('clickable');
-            pre.onclick = () => {
+            $(pre).on('click', () => {
```

Comment:
Why is all this extra jQuery needed?

Reply:
jQuery isn't strictly necessary. I think using it sparingly leads to a more readable code on certain occasions. At first, I wanted to replace `onclick` with a more flexible `addEventListener`, but then decided to opt in for jQuery's shortcut as it looks cleaner to me.

Reply:
Why would you want to remove it?

Reply:
I always lean toward removing dependencies when it's possible to do so.  Dependencies load huge blocks of code which delay the page load, and much of it isn't needed.  They accumulate as cruft.  They take maintenance work to keep up-to-date for security fixes... and in practice, that doesn't happen.  (For example, it's come up recently that `blocks.html` in this project loads two different jQuery versions, and we're guessing at what that actually does.)  jQuery in particular does some very hard work for trivial things (e.g., `$elem.css(...)`, for example) that now have much faster native implementations in modern browsers.

Though I'm not familiar with the details, the general consensus among web developers I have interacted with is that modern DOM has made many of the features of jQuery obsolete, and it's just a legacy way to do things that are now built in.  There's `querySelectorAll` to find nodes.  There are lots of new DOM functions to manipulate those nodes.  There's `map` and `forEach` and arrow functions to easily operate on entire collections of nodes in even more flexible ways than jQuery offers.  There's `addEventListener` to add multiple listeners for the same event.

Finally, in practice, CodeWorld doesn't use one or the other consistently.  There are jQuery parts, and there are native DOM parts, and that actually makes it more difficult to contribute to than either one by itself, because now you need to be comfortable with **both**.  There's a lot of appeal in making this more consistent, and for the reasons above, if I went one direction or the other, I'd want to move away from jQuery.

Here's the counter-argument, though.  We're using at least two key components (the accordion, and jqtree) that are designed around jQuery.  So that already pulls in the jQuery dependency, and relies on jQuery-isms to interact with it.  So that's a bit of a sunk cause.  Unless those components were replaced, now isn't the time to try to remove jQuery from the code.  So, I don't care enough to block development.

Honestly, the main reason I mentioned it in this pull request was that it added a bunch of changes that didn't seem to add value, and made it harder to identify the important changes for review.

Reply:
I'm not particularly attached to jQuery and not a big fan of the imperative approach. Nevertheless, despite the fact that it might be a "legacy" way of doing things, jQuery provides a lot of utility. I was suprised to learn few months ago that it still dominates web development in 2020, surpassing any other framework (such as React and Angular).

I would venture a guess that some people who are comfortable with jQuery don't have a good grasp on the underlying mechanics, while devs who can manipulate DOM with the native API can quickly pick up jQuery, if need be.

I agree that dead code does no good for the application. At the moment, nothing could be done to alleviate this concern. But it the future, once all assets are preprocessed, we will have an opportunity to eliminate any unused code from the final bundle.